### PR TITLE
[ISSUE 187] Deployment Strategy

### LINKED ISSUE #187: [ADR]: Deployment Strategy
### Description
Blocked by #222 
Decision record for the front-end and back-end deployment service

- Choosing between EC2 and Fargate for ECS cluster launch type; just EC2, using a lambda for frontend functions, and S3 buckets for frontend. 
- Beyond the CI/CD and cloud service we're using

### Approvers

Another engineer who is not working the ticket: @acouch @daphnegold @SammySteiner
@gcarson-bit (optional)
@widal001 (optional)
@lucasmbrown-usds (optional)

### Definition of Done

- [ ] The approvers for this decision have been identified (ideally before work on the ADR starts)
- [ ] The ADR is created and stored in `documentation/decisions/adr` with status "Accepted"
- [ ] The ADR has been reviewed and approved by the approvers listed above
- [ ] The ADR satisfies requirements that are outlined in the ADR template
- [ ] Any follow-up tickets have been created (if necessary)

### PR DESCRIPTION
**Pull request by daphnegold**
## Summary
Fixes #187 

## Changes proposed
ADR discussing deployment strategies considered.

### CODE REVIEW

**Review by widal001 — APPROVED** [review]
Woohoo! I think this is a smart choice, and the ADR highlights the pros and cons of the other solutions well. 

Look at all those decisionsssss 🐔🐔🐔!!! (hopefully this isn't too much of a deepcut 😅)

File: documentation/decisions/adr/00NN-deployment-service.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+# Deployment Strategy
+
+- **Status:** Accepted <!-- REQUIRED -->
+- **Last Modified:** 2023-07-20 <!-- REQUIRED -->
+- **Related Issue:** [187](https://github.com/HHS/grants-equity/issues/187) <!-- RECOMMENDED -->
+- **Deciders:** Lucas Brown, Sammy Steiner, Billy Daly <!-- REQUIRED -->
```
Comment (SammySteiner) [inline documentation/decisions/adr/00NN-deployment-service.md:6]:
```suggestion
- **Deciders:** Lucas Brown, Daphne Gold, Sammy Steiner, Billy Daly <!-- REQUIRED -->
```

File: documentation/decisions/adr/00NN-deployment-service.md [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+- **Tags:** ADR <!-- OPTIONAL -->
+
+## Context and Problem Statement
+
+We need to choose a deployment strategy for the Grants.gov modernization effort that suits our core needs and will host our entire application ecosystem, both the API and frontend layers.
+
+## Decision Drivers <!-- RECOMMENDED -->
+
+- **Reliable:** The chosen deployment strategy should be able to offer uninterrupted application uptime with consistent results.
```
Comment (SammySteiner) [inline documentation/decisions/adr/00NN-deployment-service.md:15]: AWS SLA commits to [99.99% monthly uptime](https://aws.amazon.com/compute/sla/), so let's say qualify the word `uninterrupted`. Virtually uninterrupted? or just `able to offer at least 99.99% monthly application uptime`.

File: documentation/decisions/adr/00NN-deployment-service.md [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+## Pros and Cons of the Options <!-- OPTIONAL -->
+
+### ECS with Fargate or EC2 launch type
+
+#### Fargate
+
+Fargate is an AWS serverless compute tool for containers.
+
+- **Pros**
```
Comment (SammySteiner) [inline documentation/decisions/adr/00NN-deployment-service.md:39]: Our use case (website and api) is perfect for `fargate spot` instance types which would further reduce our costs significantly. Maybe add that to the pros section?

**Review by SammySteiner — APPROVED** [review]
Looks great! Thank you for writing this up.
I left a few comments/suggestions.

**Review by lucasmbrown-usds — APPROVED** [review]
I don't have any concerns!

**Pull request merged**